File: src/models/Subject.ts
/**
 * Represents a subject.
 */
export default class Subject {
  /**
   * Creates an instance of Subject.
   *
   * @param {number | undefined} _id - The ID of the subject.
   * @param {string | undefined} _name - The name of the subject.
   * @param {string | undefined} _description - The description of the subject.
   */
  constructor(
    private _id?: number | undefined,
    private _name?: string | undefined,
    private _description?: string | undefined
  ) {}

  /**
   * Gets the ID of the subject.
   *
   * @returns {number | undefined} The ID of the subject.
   */
  get id(): number | undefined {
    return this._id;
  }

  /**
   * Sets the ID of the subject.
   *
   * @param {number | undefined} value - The ID of the subject.
   */
  set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Gets the name of the subject.
   *
   * @returns {string | undefined} The name of the subject.
   */
  get name(): string | undefined {
    return this._name;
  }

  /**
   * Sets the name of the subject.
   *
   * @param {string | undefined} value - The name of the subject.
   */
  set name(value: string | undefined) {
    this._name = value;
  }

  /**
   * Gets the description of the subject.
   *
   * @returns {string | undefined} The description of the subject.
   */
  get description(): string | undefined {
    return this._description;
  }

  /**
   * Sets the description of the subject.
   *
   * @param {string | undefined} value - The description of the subject.
   */
  set description(value: string) {
    this._description = value;
  }

  /**
   * Gets the data of the subject as an object.
   *
   * @returns {Object} An object containing the data of the subject.
   * - id: The ID of the subject.
   * - name: The name of the subject.
   * - description: The description of the subject.
   */
  getDataObject(): {
    id: number | undefined;
    name: string | undefined;
    description: string | undefined;
  } {
    return {
      id: this._id,
      name: this._name,
      description: this._description,
    };
  }
}

File: modules/models/subject.py
from typing import Dict, Union

class Subject(object):
    """Class that represents the Subject entity."""

    def __init__(self, id: Union[int, None]=None, name: Union[str, None]=None, description: Union[str, None]=None) -> None:
        """Constructor of the Subject class.

        Args:
            id (Union[int, None], optional): Unique subject record identifier. Defaults to None.
            name (Union[str, None], optional): The name of the subject. Defaults to None.
            description (Union[str, None], optional): The description of the subject. Defaults to None.
        """
        self._id = id
        self._name = name
        self._description = description
    
    @property
    def id(self) -> Union[int, None]:
        """Get the unique identifier of the subject record."""
        return self._id

    @id.setter
    def id(self, value: Union[int, None]) -> None:
        """Set the unique identifier of the subject record."""
        self._id = value

    @property
    def name(self) -> Union[str, None]:
        """Get the name of the subject."""
        return self._name

    @name.setter
    def name(self, value: Union[str, None]) -> None:
        """Set the name of the subject."""
        self._name = value

    @property
    def description(self) -> Union[str, None]:
        """Get the description of the subject."""
        return self._description
    
    @description.setter
    def description(self, value: Union[str, None]) -> None:
        """Set the description of the subject."""
        self._description = value

    def as_dict(self, include_none_data=False) -> Dict[str, Union[int, str, None]]:
        """Represents class attribute information as a data dictionary.

        Args:
            include_none_data (bool, optional): Indicates whether or not to include data with a value of None in the dict. Defaults to False.

        Returns:
            Dict[str, Union[int, str, None]]: A dictionary with the attributes as key and the corresponding value.
        """
        data = {
            'id': self._id,
            'name': self._name,
            'description': self._description
        }
        if not include_none_data:
            data = {key: value for key, value in data.items() if value is not None}

        return data